app/models/container_deployment.rb

```diff
@@ -6,7 +6,383 @@ class ContainerDeployment < ApplicationRecord
   has_many :container_volumes, :as => :parent
   has_many :custom_attributes, :as => :resource, :dependent => :destroy
   has_many :authentications, :as => :resource, :dependent => :destroy
-  serialize :customize, Hash
-
   DEPLOYMENT_TYPES = %w(origin openshift-enterprise).freeze
+  serialize :customizations, Hash
+  AUTHENTICATIONS_NAMES = {"AuthenticationAllowAll" => "AllowAllPasswordIdentityProvider", "AuthenticationHtpasswd" => "HTPasswdPasswordIdentityProvider", "AuthenticationLdap" => "LDAPPasswordIdentityProvider", "AuthenticationRequestHeader" => "RequestHeaderIdentityProvider", "AuthenticationOpenId" => "OpenIDIdentityProvider", "AuthenticationGoogle" => "GoogleIdentityProvider", "AuthenticationGithub" => "GitHubIdentityProvider", "AuthPrivateKey" => "ssh", "AuthenticationRhsm" => "rhsm"}.freeze
+  ANSIBLE_CONFIG_LOCATION = "/usr/share/atomic-openshift-utils/ansible.cfg".freeze
+  ANSIBLE_CONFIG_LOG = "/tmp/ansible.log".freeze
+  ANSIBLE_CONFIG_INVENTORY_PATH = "/tmp/inventroy.yaml".freeze
+
+  def ssh_auth
+    authentications.find_by(:type => "AuthPrivateKey")
+  end
+
+  def rhsm_auth
+    authentications.find_by(:type => "AuthenticationRhsm")
+  end
+
+  def identity_provider_auth
+    authentications.where.not(:type => "AuthPrivateKey").where.not(:type => "AuthenticationRhsm")
```

Comment:
You can collapse the `where.not` statements.  This can become:
```
authentications.where.not(:type => ["AuthPrivateKey", "AuthenticationRhsm"])
```